src/widget/woverview.cpp

```diff
@@ -661,6 +843,143 @@ void WOverview::drawCurrentPosition(QPainter* pPainter) {
     pPainter->drawLine(m_iPos - 2, breadth() - 1, m_iPos + 2, breadth() - 1);
 }
 
+void WOverview::drawTimeRuler(QPainter* pPainter) {
+    QFont markerFont = pPainter->font();
+    markerFont.setPixelSize(10 * m_scaleFactor);
+    QFontMetricsF fontMetrics(markerFont);
+
+    QFont shadowFont = pPainter->font();
+    shadowFont.setWeight(99);
+    shadowFont.setPixelSize(10 * m_scaleFactor);
+    QPen shadowPen(Qt::black, 2.5 * m_scaleFactor);
+
+    if (m_bTimeRulerActive) {
+        QLineF line;
+        if (m_orientation == Qt::Horizontal) {
+            line.setLine(m_timeRulerPos.x(), 0.0, m_timeRulerPos.x(), static_cast<float>(height()));
```

Comment:
cast to qreal or just omit the cast.

Reply:
When should qreal be used versus float or double?

Reply:
qreal should be used if you prepare a value for a function that takes a qreal. 
Depending on the QT configuration it is resolved as float or double. I we use it as well the type is correct in both cases.